### ISSUE TITLE
Problem loading vendor.js and application.js on deployed site

### ISSUE THREAD

**Issue by bradleyden** [ga-wdi-boston/full-stack-project#1025]
When I try to load my deployed site, I get 404 errors when it tries to load the vendor.js and application.js files. The urls it is trying to reach are:
https://bradleyden.github.io/project-two-front-end/public/application.js
https://bradleyden.github.io/project-two-front-end/public/vendor.js

These files load fine in the development environment like so:
http://localhost:7165/public/application.js
http://localhost:7165/public/vendor.js

Here's how the files are called in the html:
```html
      <script src="public/vendor.js" type="text/javascript" charset="utf-8" defer></script>
      <script src="public/application.js" type="text/javascript" charset="utf-8" defer></script>
```

Unfortunately, there's not much else in the error message, so I'm not sure why it can't reach them.

**Comment by jordanallain** [comment]
When you go to settings on github, what branch is selected in the Source drop down under GitHub Pages?

**Comment by payne-chris-r** [comment]
Do you have script files in your index.html?

**Comment by payne-chris-r** [comment]
```html
      <!-- Do not add `link` tags unless you know what you are doing -->
      <link rel="shortcut icon" type="image/x-icon" href="favicon.ico">
      <link href="https://fonts.googleapis.com/css?family=Open+Sans|Play" rel="stylesheet">
      <link href="https://cdn.datatables.net/1.10.15/css/dataTables.bootstrap.min.css" rel="stylesheet">
      <!-- Do not add `script` tags unless you know what you are doing -->
      <script src="https://use.fontawesome.com/b9279fc81b.js"></script>
      <script src="public/vendor.js" type="text/javascript" charset="utf-8" defer></script>
      <script src="public/application.js" type="text/javascript" charset="utf-8" defer></script>
      <script src="https://cdn.datatables.net/1.10.15/js/jquery.dataTables.min.js" type="text/javascript" charset="utf-8" defer></script>
</head>
```

**Comment by payne-chris-r** [comment]
Please read those comments ☝️

**Comment by payne-chris-r** [comment]
The approach you've chosen is not compatible with webpack. 😢 
You need to put on your google-foo gloves and try to figure out a webpack-friendly way to add those packages.

**Comment by bradleyden** [comment]
Okay, that's unfortunate. I will look into it and see if I can find a solution.

**Comment by payne-chris-r** [comment]
Sometimes constraints are put on us from a technology standpoint that are out of our control. Webpack is doing an awful lot for you, and it's benefits way outweigh the cons. I realize it may be frustrating now, but you know the code you've written works, you just need to figure out a way to incorporate those packages that will work inside browser-template (webpack)! Do a little research and then find me so we can discuss how best to move forward!

**Comment by gaand** [comment]
I get the following error going to your deployed site:

```
jquery.dataTables.min.js:5 Uncaught ReferenceError: jQuery is not defined
    at jquery.dataTables.min.js:5
    at jquery.dataTables.min.js:5
```

which is not the error you reported.  Did you change something ❓

**Comment by payne-chris-r** [comment]
I'd start here: https://www.datatables.net/download/npm (I'm don't think you'll need to redefine `$`, so try it first without).

**Comment by bradleyden** [comment]
@gaand I changed the GitHub Pages source to the gh-pages branch based on Jordan's suggestion, which seems to have fixed the issue with loading vendor.js and application.js, but is now causing the jquery error you posted.

So now my page is loading properly, but DataTables is not working, which is good because if worst comes to worst I can just get rid of DataTables and lose the nice table functionality it adds, but I think I will still be able to meet requirements with some minor retooling of the basic HTML table.

@payne-chris-r I was just looking at that link, so I will see what I can do to hopefully get DataTables to work in by tomorrow. In the meantime I can at least confirm that Heroku was deployed properly. I will find you after lunch sometime and we can discuss it!

Thanks so much for the help! I will close this issue for now, as the original issue was resolved.
